src/wallet/wallet.cpp

```diff
@@ -2991,6 +2991,165 @@ std::string CWallet::GetWalletHelpString(bool showDebug)
+        if (nZapWalletRet != DB_LOAD_OK) {
+            errorString = _("Error loading wallet.dat: Wallet corrupted");
+            uiInterface.InitMessage(_("Error loading wallet.dat: Wallet corrupted"));
+            return NULL;
+        }
+
+        delete tempWallet;
+        tempWallet = NULL;
+    }
+
+    uiInterface.InitMessage(_("Loading wallet..."));
+
+    int64_t nStart = GetTimeMillis();
+    bool fFirstRun = true;
+    CWallet *walletInstance = new CWallet(strWalletFile);
+    DBErrors nLoadWalletRet = walletInstance->LoadWallet(fFirstRun);
+    if (nLoadWalletRet != DB_LOAD_OK)
+    {
+        if (nLoadWalletRet == DB_CORRUPT)
+            errorString += _("Error loading wallet.dat: Wallet corrupted") + "\n";
+        else if (nLoadWalletRet == DB_NONCRITICAL_ERROR)
+        {
+            warningString += _("Error reading wallet.dat! All keys read correctly, but transaction data"
+                          " or address book entries might be missing or incorrect.");
+        }
+        else if (nLoadWalletRet == DB_TOO_NEW)
+            errorString += strprintf(_("Error loading wallet.dat: Wallet requires newer version of %s"), _(PACKAGE_NAME)) + "\n";
+        else if (nLoadWalletRet == DB_NEED_REWRITE)
+        {
+            errorString += strprintf(_("Wallet needed to be rewritten: restart %s to complete"), _(PACKAGE_NAME)) + "\n";
+            LogPrintf("%s", errorString);
+        }
+        else
+            errorString += _("Error loading wallet.dat") + "\n";
+
+        if (!errorString.empty())
+            return NULL;
+    }
+
+    if (GetBoolArg("-upgradewallet", fFirstRun))
+    {
+        int nMaxVersion = GetArg("-upgradewallet", 0);
+        if (nMaxVersion == 0) // the -upgradewallet without argument case
+        {
+            LogPrintf("Performing wallet upgrade to %i\n", FEATURE_LATEST);
+            nMaxVersion = CLIENT_VERSION;
+            walletInstance->SetMinVersion(FEATURE_LATEST); // permanently upgrade the wallet immediately
+        }
+        else
+            LogPrintf("Allowing wallet upgrade up to %i\n", nMaxVersion);
+        if (nMaxVersion < walletInstance->GetVersion())
+        {
+            errorString += _("Cannot downgrade wallet") + "\n";
+            return NULL;
+        }
+        walletInstance->SetMaxVersion(nMaxVersion);
+    }
+
+    if (fFirstRun)
+    {
+        // Create new keyUser and set as default key
+        RandAddSeedPerfmon();
+
+        CPubKey newDefaultKey;
+        if (walletInstance->GetKeyFromPool(newDefaultKey)) {
+            walletInstance->SetDefaultKey(newDefaultKey);
+            if (!walletInstance->SetAddressBook(walletInstance->vchDefaultKey.GetID(), "", "receive"))
+            {
+                errorString += _("Cannot write default address") += "\n";
+                return NULL;
+            }
+        }
+
+        walletInstance->SetBestChain(chainActive.GetLocator());
+    }
+
+    LogPrintf("%s", errorString);
```

Comment:
Right. This is dead code now. Move the log print to the init.cpp error handling (L1434).